### ISSUE TITLE
Bit fields all displaying as 0

### ISSUE THREAD

**Issue by ghost** [sequelpro/sequelpro#749]
Issue imported from Google Code: https://code.google.com/p/sequel-pro/issues/detail?id=749
Reporter:richard....@gmail.com Date:2010-06-28 11:50:28 Status:Fixed Closed:2010-07-01

Please forgive me if I'm doing something wrong; I'm new to both MySQL and Sequel Pro. However, to the best of my knowledge (having checked all the available documentation etc.), the issue I've found is a genuine one. This is in Sequel Pro version 0.9.8 and MySQL version 5.1.43.

In one of my tables I have a BIT field, which is defined to hold 10 (ten) bits. I would expect to be able to both enter binary values and see the stored results as meaningful binary values in Sequel Pro, but its interface to BIT files seems to be completely non-working.

If I enter any value other than NULL (in whatever format) directly into the table, then on updating the row, only a single 0 is displayed.

However, if I go to the Query window and issue an explicit UPDATE command to insert a binary value into that particular field (in the form 0b1010 or b'1010'), the command is accepted... but the field in question still displays just a 0 when I check the table view.

If I then go to the structure view and convert the BIT field into (say) a SMALLINT field, then look again at the values in the table, what I find is that:

(a) Any values I've inserted via typed UPDATE commands in the Query window show the correct decimal equivalent, despite the fact that they displayed as 0 when the field was of type BIT. For example, a 0b11000 entry now shows as 24; a 0b1001 show as 9, etc. So MySQL is clearly inserting the correct values into the table, but Sequel Pro is showing every value (other than NULL) as 0.

(b) Any values I've inserted directly into the table view, despite showing as 0 after I'd entered them, now show as 1 once converted into SMALLINT. (If I convert back to BIT, all values show as 0 again.)

To summarise, if I try to work with BIT fields in Sequel Pro, then:
- All values I try to enter are entered as 1.
- All values actually stored are displayed as 0.

The correct values are clearly present in the table, because converting between BIT and SMALLINT converts between the incorrect BIT displays and the actual decimal equivalents of the binary values I stored via UPDATE commands.

In my application, I can work around this problem by using decimal values rather than binary ones, but it would be preferable for the BIT field to work as presumably it should.
